cl/alerts/tests/tests.py

```diff
@@ -2319,6 +2322,284 @@ def test_long_alert_subjects_truncation(self):
+        )
+
+        # Validate payload structure
+        for event in webhook_events:
+            content = event.content
+            self.assertEqual(
+                content["webhook"]["event_type"],
+                WebhookEventType.PRAY_AND_PAY,
+            )
+            self.assertEqual(content["payload"]["status"], Prayer.GRANTED)
+            self.assertEqual(content["payload"]["recap_document"], rd.id)
+            self.assertIn(
+                content["payload"]["id"], [prayer_v1.id, prayer_v2.id]
+            )
+
+        # Confirm webhooks for V1 and V2 are properly triggered
+        webhook_versions = {
+            event.content["webhook"]["version"] for event in webhook_events
+        }
+        self.assertEqual(webhook_versions, {1, 2})
+
+    def test_pray_and_pay_webhook_payload_fields(self):
+        """Test detailed field validation for pray-and-pay webhooks.
+
+        Verifies the payload matches the expected structure through the
+        complete signal flow.
+        """
+        from cl.favorites.factories import PrayerFactory
+        from cl.favorites.models import Prayer
+
+        # Create unavailable document and waiting prayer
+        de = DocketEntryFactory(docket__court=self.court, entry_number=2)
+        rd = RECAPDocumentFactory(
+            docket_entry=de,
+            document_number=2,
+            is_available=False,
+        )
+
+        prayer = PrayerFactory(
+            user=self.user_profile.user,
+            recap_document=rd,
+            status=Prayer.WAITING,
+        )
+
+        with mock.patch(
+            "cl.api.webhooks.requests.post",
+            side_effect=lambda *args, **kwargs: MockResponse(
+                200, mock_raw=True
+            ),
+        ):
+            # Trigger signal by making document available
+            rd.is_available = True
+            rd.save()
+
+        webhook_event = WebhookEvent.objects.first()
+        content = webhook_event.content
+        payload = content["payload"]
+
+        # Validate all required fields are present
+        self.assertIn("id", payload)
+        self.assertIn("date_created", payload)
+        self.assertIn("status", payload)
+        self.assertIn("recap_document", payload)
+
+        # Validate field values
+        self.assertEqual(payload["id"], prayer.id)
+        self.assertEqual(payload["status"], Prayer.GRANTED)
+        self.assertEqual(payload["recap_document"], rd.id)
+
+        # Validate webhook metadata
+        self.assertEqual(
+            content["webhook"]["event_type"],
+            WebhookEventType.PRAY_AND_PAY,
+        )
+
+    def test_pray_and_pay_webhook_only_granted_status(self):
+        """Test that webhooks are only sent when prayers are granted.
+
+        Documents that are already available should not trigger webhooks
+        when prayers are created.
+        """
+        from cl.favorites.factories import PrayerFactory
+        from cl.favorites.models import Prayer
```

Comment:
And here too. Sorry, I didn’t highlight them in my first pass since they’re identical.